pkg/master/controller.go

```diff
@@ -83,6 +86,11 @@ func (c *Controller) Start() {
 	repairClusterIPs := servicecontroller.NewRepair(c.ServiceClusterIPInterval, c.ServiceRegistry, c.ServiceClusterIPRange, c.ServiceClusterIPRegistry)
 	repairNodePorts := portallocatorcontroller.NewRepair(c.ServiceNodePortInterval, c.ServiceRegistry, c.ServiceNodePortRange, c.ServiceNodePortRegistry)
 
+	// Initialize the master leases
+	if c.MasterLeases != nil {
```

Comment:
Ah, that's right, I just wanted to automatically have the lease time be directly based on the endpoint interval, without having to separately pass it to the constructor (to remove the chance of having one accidentally set without setting the other).  It does make the actual constructor look a bit weird though.  I'm fine with just changing it to be set when the controller is constructed if you'd rather have that, though.